src/components/NetworkError/NetworkError.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect, useState } from 'react';
+import './NetworkError.scss';
+
+export const NetworkError: React.FC = () => {
+  const [isOnline, setOnline] = useState<boolean>(navigator.onLine);
+
+  useEffect(() => {
```

Comment:
This code also follows Single Responcibility Principle: as it is only responcible for rendering an error message when the user is offline.